dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/IdentifiableObjectManager.java

```diff
@@ -81,8 +82,17 @@
     @Nonnull
     Optional<? extends IdentifiableObject> find( @Nonnull String uid );
 
+    /**
+     *
+     * @param type the object class type.
+     * @param id object's database ID
+     * @return the found object
+     * @throws ObjectNotFoundException when no such object exists in the
```

Comment:
I think this was the behaviour all along as this should be how hibernate responds to lookup by ID when the object does not exist. This should just make this clear.